src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -102,6 +103,20 @@ function checkAndWarnForMutatedStyle(style1, style2, component) {
   if (style1 == null || style2 == null) {
     return;
   }
+
+  // Remove any NaN values from the style objects before the shallow equal check
+  // so they don't give a false positive
+  for (var key1 in style1) {
+    if (numberIsNaN(style1[key1])) {
+      delete style1[key1];
```

Comment:
I still feel like we probably shouldn't be modifying the argument.

This is a dev-only code path, so maybe we copy the object and mutate the copy.  Or maybe we fork shallowEqual to have the logic we want with regards to NaN in this case.  Or maybe there is another option/thought.

cc @sebmarkbage @spicyj